pkg/k8s/apps/translate.go

```diff
@@ -160,30 +158,6 @@ func TranslateDevTolerations(spec *apiv1.PodSpec, tolerations []apiv1.Toleration
 	spec.Tolerations = append(spec.Tolerations, tolerations...)
 }
 
-// TranslatePodAffinity translates the affinity of pod to be all on the same node
-func TranslatePodAffinity(spec *apiv1.PodSpec, name string) {
```

Comment:
I moved this logic to `ToTranslationRule`. Translations rules already supports affinities. We can inject the affinities directly in the translation rule. This way we have more context to make this decision (for example, we have access to all the PersistentVolume fields).